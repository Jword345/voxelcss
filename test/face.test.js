'use strict';

const ColorFace = require('../lib/ColorFace');
const ImageFace = require('../lib/ImageFace');

describe('ColorFace', function() {
    let face;
    beforeEach(() => {
        face = new ColorFace('#333');
    });
    it('can be created without color parameter', () => {
        face = new ColorFace();
        expect(face.getColor()).toMatchSnapshot();
    });
    it('can get hex and RGBA color values', () => {
        expect(face.getHex()).toMatchSnapshot();
        expect(face.getColor()).toMatchSnapshot();
    });
    it('can be cloned', () => {
        let newFace = face.clone();
        expect(newFace.getColor()).toMatchSnapshot();
    });
});
describe('ImageFace', function() {
    let face;
    beforeEach(() => {
        face = new ImageFace();
    });
    it('can be created with default image', function() {
        expect(face.getSource()).toMatchSnapshot();
    });
    it('can be cloned', function() {
        let newFace = face.clone();
        expect(newFace.getSource()).toMatchSnapshot();
    });
});
