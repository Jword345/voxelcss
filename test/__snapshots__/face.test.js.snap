// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorFace can be cloned 1`] = `
Object {
  "a": 1,
  "b": 51,
  "g": 51,
  "r": 51,
}
`;

exports[`ColorFace can be created without color parameter 1`] = `
Object {
  "a": 1,
  "b": 0,
  "g": 0,
  "r": 0,
}
`;

exports[`ColorFace can get hex and RGBA color values 1`] = `"333333"`;

exports[`ColorFace can get hex and RGBA color values 2`] = `
Object {
  "a": 1,
  "b": 51,
  "g": 51,
  "r": 51,
}
`;

exports[`ImageFace can be cloned 1`] = `"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"`;

exports[`ImageFace can be created with default image 1`] = `"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"`;
